src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1214,6 +1219,16 @@ object SymDenotations {
       myTypeParams
     }
 
+    /** The named type parameters declared or inherited by this class */
+    override final def namedTypeParams(implicit ctx: Context): Set[TypeSymbol] = {
+      def computeNamedTypeParams: Set[TypeSymbol] =
+        if (ctx.erasedTypes || is(Module)) Set() // fast return for modules to avoid scanning package decls
+        else memberNames(abstractTypeNameFilter).map(name =>
```

Comment:
Shouldn't we create a `NameFilter` instance that filters named type params directly instead?
```scala
  object namedTypeParamNameFilter extends NameFilter {
    def apply(pre: Type, name: Name)(implicit ctx: Context): Boolean =
      name.isTypeName && {
        val mbr = pre.member(name)
        mbr.symbol.is(Deferred) && mbr.symbol.is(TypeParam, butNot = ExpandedName) &&
        mbr.info.isInstanceOf[RealTypeBounds]
      }
  }
```